src/bridge/transactions/signing.rs

```diff
@@ -300,3 +314,20 @@ pub fn populate_taproot_input_witness_default(
         script,
     );
 }
+
+pub fn populate_taproot_input_witness_with_signature(
+    tx: &mut Transaction,
+    input_index: usize,
+    taproot_spend_info: &TaprootSpendInfo,
+    script: &Script,
+    signatures: &Vec<bitcoin::taproot::Signature>,
+) {
+    let unlock_data = signatures.iter().map(|sig| sig.to_vec()).collect();
+    push_taproot_leaf_unlock_data_to_witness(tx, input_index, unlock_data);
+    push_taproot_leaf_script_and_control_block_to_witness(
+        tx,
+        input_index,
+        taproot_spend_info,
+        script,
+    );
```

Comment:
Just call `populate_taproot_input_witness` function instead
```suggestion
    populate_taproot_input_witness(
        tx,
        input_index,
        taproot_spend_info,
        script,
        unlock_data,
    );
```

Reply:
Fixed as suggested